presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -2045,6 +2073,43 @@ private String typeParameterToString(SqlBaseParser.TypeParameterContext typePara
         throw new IllegalArgumentException("Unsupported typeParameter: " + typeParameter.getText());
     }
 
+    private Optional<GrantorSpecification> getGrantorSpecificationIfPresent(SqlBaseParser.GrantorContext context)
+    {
+        return Optional.ofNullable(context).map(this::getGrantorSpecification);
+    }
+
+    private GrantorSpecification getGrantorSpecification(SqlBaseParser.GrantorContext context)
```

Comment:
I do not like this `instanceof`s.

Possible options:
1. We can make `GrantorSpecification` and `PrincipalSpecification` subclasses of `Node`. And add specific `visitSpecifiedPrincipal` ... methods to AST builder. And use `visitIfPresent(.., GrantorSpecification.class` instead `getGrantorSpecificationIfPresent`.

2. We can instantiate small visitor (specific subclass of `SqlBaseBaseVisitor`) just for handling `Grantor` within `getGrantorSpecification`

Reply:
+1

See how it is handle `levelOfIsolation` there is nested `visit` (Visitor) used, no ifs at all. Lets follow the current convention

Reply:
ignored for now